src/components/MealsList.js

```diff
@@ -1,30 +1,53 @@
 import { Box, Divider } from "@mui/material";
 import MealsItem from "./MealsItem";
 import NavigationPannel from "./NavigationPannel";
+import Tags from "./Tags";
+import { useState } from "react";
 
 export default function MealsList({
   meals,
   handlePlanBtn,
   handleOpenMealBtnClick,
+  allIngridients,
 }) {
+  const [search, setSearch] = useState([]);
+
+  const filteredMeals = meals.filter((meal) => {
+    const ingridients = meal.ingridients;
+    let index = 0;
+    search.forEach((value) => {
```

Comment:
Instead of using index and 2 `forEach` you can use other methods.

```
const areAllFound = search.every(si => {
  return si.some(i => {
    return i.toLowerCase() === si.toLowerCase()
  })
})
```

(short names for sake of typing less)